### PR TITLE
feat(CX-3618): create the view all artwork grid page for auctions works for you

### PR DESCRIPTION
**Pull request by dariakoko**
The type of this PR is: **Feat**

This PR solves [CX-3618]

### Description

image

https://user-images.githubusercontent.com/36167539/233123247-7b898976-616e-4f83-bbb5-fa2bcdaa5517.mov

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CX-3618]: https://artsyproduct.atlassian.net/browse/CX-3618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Routes added to `routes.tsx` should have a corresponding cypress.js smoke test. See the `cypress/e2e` folder for examples.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 3462420ef2c22042452de1fc218ee5fcd1e02882
</p>

File: src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          <Message variant="warning">
+            <Clickable onClick={handleClick} textDecoration="underline">
+              Log in
+            </Clickable>{" "}
+            to see your personalized recommendations.
+          </Message>
+
+          <Spacer y={4} />
+        </>
+      )}
```
Comment (olerichter00) [inline src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx:54]: **suggestion:** I wonder if we already have a component for this message we could use 🤔

File: src/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx [hunk 2]
```diff
@@ -11,6 +11,10 @@ interface NewForYouArtworksGridProps {
 export const NewForYouArtworksGrid: FC<NewForYouArtworksGridProps> = ({
   viewer,
 }) => {
+  console.log(
+    "[LOGD] viewer.artworksForUser.totalCount = ",
+    viewer.artworksForUser?.totalCount
+  )
```
Comment (olerichter00) [inline src/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx:17]: 👀 

```suggestion
```

File: src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx (hunk 1 above)
Comment (dariakoko) [reply src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx:54]: We don't, but this is the second place where this design is used. I guess we can create one. Will do 👍 

I'm never sure with such small components, I feel like when we create one it is often missed. Especially if the component is not in palette. Maybe it's just me 👀

File: src/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx (hunk 2 above)
Comment (dariakoko) [reply src/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx:17]: 👀 💧

File: src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx (hunk 1 above)
Comment (olerichter00) [reply src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx:54]: I like the idea of creating a separate component. If we already use it in two places, it makes a lot of sense.

And the component is probably easier to read.

File: src/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx (hunk 2 above)
Comment (olerichter00) [reply src/Apps/NewForYou/Components/NewForYouArtworksGrid.tsx:17]: 🖨️

File: src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx (hunk 1 above)
Comment (dariakoko) [reply src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx]: Done!

**Comment by dariakoko** [comment]
With @olerichter00 we discovered that the problem with pagination above is a common one and it's better to be addressed globally

File: src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    <>
+      <MetaTags title="Auction Lots for You Ending Soon" />
+
+      <Spacer y={4} />
+
+      <Text variant="xl">Auction Lots for You Ending Soon</Text>
+
+      <Spacer y={4} />
+
+      <LogInPrompt contextModule={ContextModule.auctionLotsEndingSoonRail} />
```
Comment (olerichter00) [inline src/Apps/Auctions/EndingSoonAuctions/EndingSoonAuctions.tsx:24]: 👍

**Review by olerichter00 — APPROVED** [review]
Looks great 🌟

**Comment by MounirDhahri** [comment]
Glad you figured it out together

**Review by MounirDhahri — APPROVED** [review]

**Pull request merged**